library/nsxt_upgrade_prechecks.py

```diff
@@ -0,0 +1,178 @@
+                                  mgr_password, validate_certs, time_out=10800):
+  '''
+    params:
+    - endpoint: API endpoint.
+    - attribute_list: The attribute whose value should become the desired attribute value
+    - desired_attribute_value: The desired attribute value
+    
+    Function will wait till the attribute value derived from going deep to attribute list
+    becomes equal to desired_attribute_value.
+   '''
+  operation_time = 0
+  while True:
+    try:
+      (rc, resp) = request(manager_url + endpoint, headers=dict(Accept='application/json'),
+                           url_username=mgr_username, url_password=mgr_password, 
+                           validate_certs=validate_certs, ignore_errors=True)
+    except Exception as err:
+       pass
+    if resp.__contains__('component_status'):
+      flag = True
+      component_statuses = resp['component_status']
+      for component_status in component_statuses:
+        if component_status['pre_upgrade_status']['status'] != 'COMPLETED':
+          flag = False
+      if flag:
+        return None
+    time.sleep(15)
+    operation_time = operation_time + 15
+    if operation_time > time_out:
+      raise Exception('Operation timed out.')
+
+def main():
+  argument_spec = vmware_argument_spec()
+  argument_spec.update(timeout=dict(type='int', required=False),
+                      state=dict(required=True, choices=['present', 'absent']))
+
+  module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+  upgrade_prechecks_params = clean_and_get_params(module.params.copy(), ['timeout'])
+  state = module.params['state']
+  mgr_hostname = module.params['hostname']
+  mgr_username = module.params['username']
+  mgr_password = module.params['password']
+  validate_certs = module.params['validate_certs']
+  timeout = module.params['timeout']
+
+  headers = dict(Accept="application/json")
+  headers['Content-Type'] = 'application/json'
+  
+  mgr_hostname = get_upgrade_orchestrator_node(module, mgr_hostname, mgr_username, 
+                                            mgr_password, headers, validate_certs)
+
+  manager_url = 'https://{}/api/v1'.format(mgr_hostname)
+
+  if state == 'present':
+    # Runs pre upgrade checks
+    if module.check_mode:
+      module.exit_json(changed=False, debug_out='Pre upgrade checks will be executed.', 
+                       id='Pre upgrade checks')
+    request_data = json.dumps(upgrade_prechecks_params)
+    try:
+      (rc, resp) = request(manager_url + '/upgrade?action=execute_pre_upgrade_checks', 
+                           data='', headers=headers, method='POST', 
+                           url_username=mgr_username, url_password=mgr_password, 
+                           validate_certs=validate_certs, ignore_errors=True)
+    except Exception as err:
+      module.fail_json(msg="Failed to execute pre upgrade checks. Error[%s]." % to_native(err))
+
+    try:
+      if timeout is None:
```

Comment:
see comment in post_check module

Reply:
This is required. We need to send timeout parameter. Can't send None value. Thus this check.